### ISSUE TITLE
lr_scheduler order changed since PyTorch 1.1.0

### ISSUE THREAD

**Issue by yongduek** [pytorchbearer/torchbearer#660]
Many thanks for the wonderful library. 

A warning message emerged when a scheduler was used in the callbacks:
`
scheduler = torchbearer.callbacks.torch_scheduler.StepLR(step_size=5, gamma=0.1)
`

Hope it could be considered in later update if not yet included.

```
Python37\lib\site-packages\torch\optim\lr_scheduler.py:100: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

**Comment by kikatuso** [comment]
Just wanted to let you know, that I am still getting the error mentioned above. My code: 

```
from torchbearer.callbacks import StepLR
optimiser = optim.Adam(filter(lambda p: p.requires_grad, model.parameters()), lr=0.001)
loss_function = nn.CrossEntropyLoss()

device = "cuda:0" if torch.cuda.is_available() else "cpu"
scheduler = StepLR(step_size=2, gamma=0.1)
trial = Trial(model, optimiser, loss_function,callbacks=[scheduler], metrics=['loss', 'accuracy']).to(device)
trial.with_generators(train_loader, val_generator=val_loader, test_generator=test_loader)
history=trial.run(epochs=10)
results = trial.evaluate(data_key=torchbearer.TEST_DATA)
print(results)
```

I am getting these two warnings:

> /usr/local/lib/python3.7/dist-packages/torch/optim/lr_scheduler.py:134: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
>   "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
> 
>  /usr/local/lib/python3.7/dist-packages/torch/optim/lr_scheduler.py:154: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
>   warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)

**Comment by ethanwharris** [comment]
Hi @kikatuso, thanks for your comment! This should now be fixed if you install from master (use the build from source instructions here: https://github.com/pytorchbearer/torchbearer#install). I should point out that `torchbearer` is not currently in active development and so may not be very reliable. Consider switching to an alternative like [`pytorch-lightning`](https://github.com/PyTorchLightning/pytorch-lightning).

Thanks :smiley:
